packages/frontend/src/pages/EpochKeyPage.jsx

```diff
@@ -25,67 +26,30 @@ export default observer(() => {
   const attestations = unirep.attestationsByEpochKey.get(epochKeyId)
 
   return (
-    <>
+    <div className="content">
+      <Header />
       <div className="container">
         <div className="left-container">
           <h3>Epoch Key</h3>
-          <div className="info-card">
-            {attestations ? (
-              <>
-                <div className="flex">
-                  <h5>Attester</h5>
-                  <Link
-                    to={`/attester/0x${BigInt(
-                      attestations[0].attesterId
-                    ).toString(16)}`}
-                  >
-                    <h4>
-                      <span>
-                        {`0x${BigInt(attestations[0].attesterId).toString(
-                          16
-                        )}`.slice(0, 7)}
-                      </span>
-                      ...
-                      <span>
-                        {`0x${BigInt(attestations[0].attesterId).toString(
-                          16
-                        )}`.slice(-5)}
-                      </span>
-                    </h4>
-                  </Link>
-                </div>
-                <div className="flex">
-                  <h5>Epoch</h5>
-                  <h6>{attestations[0].epoch}</h6>
-                </div>
-                <div className="flex">
-                  <h5>Attestations Received</h5>
-                  <h6>{attestations.length}</h6>
-                </div>
-                <div className="flex">
-                  <h5>Epoch Key</h5>
-                  <Tooltip
-                    text="A user's temporary, random-value-like epochKey is the receiver of attestations."
-                    maxWidth={200}
-                  />
-                </div>
-                <h6 className="break">{id}</h6>
-              </>
-            ) : (
-              'Loading...'
-            )}
-          </div>
+          {attestations ? (
+            <EpochKeyInfo
```

Comment:
I created the `EpochKeyInfo` component because I needed to call `loadEpochsByAttester(attesterId)` AFTER getting the `attesterId` from the epoch key's `attestations` that are loaded in `EpochKeyPage`. So the extra component is the solution I came up with; I get the `attestations` first, then pass the `attesterId` to `EpochKeyInfo` so the component asks for the relevant attester's epoch info. (actually all of this is just to find if the epoch is sealed or not, the rest of the needed data is contained in the attestation object)

Thanks for sharing these guidelines for component use. If the above isn't an acceptable reason to keep this component:
- likely there is a solution that I didn't discover for calling both loading functions from the same component; any ideas?
- is there a better way to get this piece of data - like would it be possible to add `epochSealed?` to the attestation object?

Reply:
Ok, gotcha, yes there's definitely a better approach to this. To me what you're describing is a _data_ problem. Usually a data problem is not solved at the ui level, a good solution might be in the stores, or in the backend.

I would solve this by changing the return data for `/api/attester/:attesterId/attestations`. In this route first load all the attestations, then load all the epoch objects for the attestations. You can select in the database by matching against a list of values by providing an array in the `where` clause. So something like this

```js

    const { attesterId } = req.params
    const attestations = await db.findMany('Attestation', {
      where: {
        attesterId: attesterId,
      },
    })
    const epochs = await db.findMany('Epoch', {
      where: {
        attesterId,
        number: attestations.map(a => a.epoch)
      }
    })
```

The [schema docs](https://developer.unirep.io/docs/next/core-api/schema) should be helpful.

In the [existing implementation](https://github.com/Unirep/explorer/blob/review-updates/packages/backend/src/routes/attestationsByAttester.mjs#L13) you return an array from the route. Instead you should return an object like
```js
res.json({ items })
```

If you return an object it's easier to add more data that is related like this

```js
res.json({ items, epochs })
```

Then you can ingest both sets of data in the stores and have it synchronously available in the ui.